src/uthash.h

```diff
@@ -1172,10 +1176,16 @@ typedef struct UT_hash_bucket {
 
 typedef struct UT_hash_table {
    UT_hash_bucket *buckets;
+#ifdef DUTHASH_V2_ABI
```

Comment:
I see why you did this, but actually I meant `UTHASH_V2_ABI` to be the *current/old* ABI (because the current uthash version number is 2.x). Anyway, I've got a more future-proof(?) idea now:

    #ifndef UTHASH_ABI_VERSION
    #define UTHASH_ABI_VERSION 3
    #endif

    #if UTHASH_ABI_VERSION >= 3
        tail, hho, num_buckets, log2_num_buckets, num_items
    #else
        num_buckets, log2_num_buckets, num_items, tail, hho
    #endif

This would make it easier to jump again to V4, V5,... if we needed to, while ensuring that anyone who needed ABI stability would only ever have to learn one pattern: `-DUTHASH_ABI_VERSION=(whatever today's major version number is)`.

Reply:
+1 for `UTHASH_ABI_VERSION`. We should upgrade the `UTHASH_VERSION` to `3.0.0` too, what do you thing?